Dockerfile

```diff
@@ -4,16 +4,17 @@ RUN apt update \
     && apt install -y libclang-dev clang \
         build-essential tcl protobuf-compiler file \
         libssl-dev pkg-config git\
-    && apt clean \
-    && cargo install cargo-chef
+    && apt clean && rm -rf /var/lib/apt/lists/*
+
 # We need to install and set as default the toolchain specified in rust-toolchain.toml
 # Otherwise cargo-chef will build dependencies using wrong toolchain
 # This also prevents planner and builder steps from installing the toolchain over and over again
 COPY rust-toolchain.toml rust-toolchain.toml
 RUN cat rust-toolchain.toml | grep "channel" | awk '{print $3}' | sed 's/\"//g' > toolchain.txt \
     && rustup update $(cat toolchain.txt) \
     && rustup default $(cat toolchain.txt) \
-    && rm toolchain.txt rust-toolchain.toml
+    && rm toolchain.txt rust-toolchain.toml \
+    && cargo install cargo-chef
```

Comment:
This is actually a regression. If rust-toolchain.toml changes `cargo install cargo-chef` will be re-run. Previously it would be cached.

Reply:
If the toolchain changes we'll have to rebuild everything anyway, the cost of running `cargo install cargo-chef` is marginal.
I think the benefit of having all rust tools built by the same toolchain outweighs the downsides.
Let me know if you still want me to move it back.

Reply:
That's fine. Just don't name the commit "Optimize ... layers" when in fact you're de-optimizing it.
If you have named your intend of "cleaning up" the layers, I wouldn't add such comment at all.